src/rapaio/math/linear/dense/CholeskyDecomposition.java

```diff
@@ -47,22 +47,9 @@ public static CholeskyDecomposition from(RM a) {
 
     private static final long serialVersionUID = -3047433451986241586L;
 
-    /**
-     * Array for internal storage of decomposition.
-     */
-    private final double[][] L;
-
-    /**
-     * Row and column dimension (square matrix).
-     */
-    private int n;
+    private CholeskyDecompositionData data = new CholeskyDecompositionData();
```

Comment:
For CholeskyDecomposition the whole idea of keeping an internal double[][] variable is to make is fast. Otherwise we do not need a new data structure to hold the matrix, since there is one already (named SolidRM). So, either we use double[][] inside and leave it like it is, either we use SolidRM. There is no need a new matrix structure.